### PR TITLE
[Material Toolbar] Change logo scaleType and set logo adjustViewBounds

### PR DESCRIPTION
**Pull request by P1NG2WIN**
- [x] Link to GitHub issues it solves. https://github.com/material-components/material-components-android/issues/2405 

Actually not solving _set shape appearance_. Didn't work with it, so I don't know how to set and update it programmatically. Planned to override setLogo(Drawable) and create MaterialShapeDrawable  from it + shapeappearance

### CODE REVIEW

**Comment by P1NG2WIN** [comment]
@googlebot I signed it!

File: lib/java/com/google/android/material/appbar/MaterialToolbar.java [hunk 1]
```diff
@@ -73,6 +74,20 @@
   private boolean titleCentered;
   private boolean subtitleCentered;
 
+  private ImageView.ScaleType logoScaleType;
+  private boolean logoAdjustViewBounds;
+
+  private static final ImageView.ScaleType[] sScaleTypeArray = {
```
Comment (drchen) [inline lib/java/com/google/android/material/appbar/MaterialToolbar.java:80]: LOGO_SCALE_TYPE_ARRAY, move it to next to other static final variables.

File: lib/java/com/google/android/material/appbar/MaterialToolbar.java [hunk 2]
```diff
@@ -97,6 +112,13 @@ public MaterialToolbar(@NonNull Context context, @Nullable AttributeSet attrs, i
     titleCentered = a.getBoolean(R.styleable.MaterialToolbar_titleCentered, false);
     subtitleCentered = a.getBoolean(R.styleable.MaterialToolbar_subtitleCentered, false);
 
+    final int index = a.getInt(R.styleable.MaterialToolbar_logoScaleType, 3);
+    if (index >= 0) {
+      logoScaleType = (sScaleTypeArray[index]);
```
Comment (drchen) [inline lib/java/com/google/android/material/appbar/MaterialToolbar.java:117]: nit) parenthesis not needed

File: lib/java/com/google/android/material/appbar/MaterialToolbar.java [hunk 3]
```diff
@@ -97,6 +112,13 @@ public MaterialToolbar(@NonNull Context context, @Nullable AttributeSet attrs, i
     titleCentered = a.getBoolean(R.styleable.MaterialToolbar_titleCentered, false);
     subtitleCentered = a.getBoolean(R.styleable.MaterialToolbar_subtitleCentered, false);
 
+    final int index = a.getInt(R.styleable.MaterialToolbar_logoScaleType, 3);
+    if (index >= 0) {
```
Comment (drchen) [inline lib/java/com/google/android/material/appbar/MaterialToolbar.java:116]: I think we should check upper bound instead. I think index is always larger than or equal to 0 here.

File: lib/java/com/google/android/material/appbar/res/values/attrs.xml [hunk 4]
```diff
@@ -272,6 +272,35 @@
          should be centered horizontally within the toolbar. Default is false.
     -->
     <attr name="subtitleCentered" format="boolean" />
+    <!-- adjust view bounds for logo's imageView. Default is false.   -->
+    <attr name="logoAdjustViewBounds" format="boolean" />
```
Comment (drchen) [inline lib/java/com/google/android/material/appbar/res/values/attrs.xml:276]: Update the doc and public.xml as well.

Comment (P1NG2WIN) [reply lib/java/com/google/android/material/appbar/res/values/attrs.xml:276]: Where doc?

**Comment by P1NG2WIN** [comment]
Where doc?

File: lib/java/com/google/android/material/appbar/res/values/attrs.xml (hunk 4 above)
Comment (drchen) [reply lib/java/com/google/android/material/appbar/res/values/attrs.xml:276]: https://github.com/material-components/material-components-android/blob/master/docs/components/TopAppBar.md

**Comment by P1NG2WIN** [comment]
update docs

**Comment by P1NG2WIN** [comment]
@drchen done

File: lib/java/com/google/android/material/appbar/MaterialToolbar.java (hunk 1 above)
Comment (drchen) [reply lib/java/com/google/android/material/appbar/MaterialToolbar.java:73]: Please fix the name. Static final variables should be ALL_CAPS

File: lib/java/com/google/android/material/appbar/MaterialToolbar.java [hunk 5]
```diff
@@ -97,6 +112,11 @@ public MaterialToolbar(@NonNull Context context, @Nullable AttributeSet attrs, i
     titleCentered = a.getBoolean(R.styleable.MaterialToolbar_titleCentered, false);
     subtitleCentered = a.getBoolean(R.styleable.MaterialToolbar_subtitleCentered, false);
 
+    final int index = a.getInt(R.styleable.MaterialToolbar_logoScaleType, 3);
+    logoScaleType = (sScaleTypeArray[index]);
```
[the remark this answers is not in the document]
Comment (drchen) [reply lib/java/com/google/android/material/appbar/MaterialToolbar.java]: Please fix

**Review by drchen — CHANGES_REQUESTED** [review]

**Comment by P1NG2WIN** [comment]
@drchen fix what? i change to this in last pr
image

**Comment by drchen** [comment]
Sorry somehow I didn't see the change yesterday. Please fix the name of sScaleTypeArray and I can try to push it for internal review.

**Comment by P1NG2WIN** [comment]
what is wrong with name?

**Comment by drchen** [comment]
Please use ALL_CAPS, which is the code convention of the Material library.

**Review by drchen — APPROVED** [review]

**Comment by drchen** [comment]
Hi, I think you may need to merge your branch with master again.

**Pull request merged**